entities/entity_class.gd

```diff
@@ -45,13 +50,60 @@ func do_collect(entity: Entity):
 			status_sight_timer.one_shot = true
 			status_sight_timer.timeout.connect(func():
 				ss_node.hide()
-				status_sight = false
+				_status_sight = false
 			)
 			add_child(status_sight_timer)
 		status_sight_timer.start()
-		status_sight = true
+		_status_sight = true
 		ss_node.show()
+	if entity is Chest:
+		if gold_pocket > 2:
+			print(gold_pocket, "Error, setting gold to 0") 
+			gold_pocket = 0
+		gold_pocket += 1
+		print(gold_pocket)
+		if gold_pocket == 2:
+			gold_pocket = 0
+			gold_status_function()
+			
+			
 
+			
+func get_sg_node() -> Node2D:
+	var node_name := "StatusGold"
+	var sg_node := find_child(node_name)
+
+	if sg_node == null:
+		var status_gold_scene := preload("res://status_gold.tscn") as PackedScene
+		var status_gold_instance := status_gold_scene.instantiate()
+		status_gold_instance.name = node_name
+		add_child(status_gold_instance)
+		status_gold_instance.show()
+		return status_gold_instance
+	
+	return sg_node
+
+func gold_status_function() -> void:
+	var sg_node: Node2D = get_sg_node()
+	print(sg_node, "gold_status_function")
+		
+	if !status_gold_timer:
+		print(gold_pocket, "Timer")
+		status_gold_timer = Timer.new()
+		status_gold_timer.wait_time = 2
+		status_gold_timer.one_shot = true
+		status_gold_timer.timeout.connect(func():
+			sg_node.hide()
+			_status_gold = false
+			gold_pocket = 0
+		)
+		add_child(status_gold_timer)
+		status_gold_timer.start()
+		_status_gold = true
+		sg_node.show()
```

Comment:
if you take these 3 lines:
```gdscript
status_gold_timer.start()
_status_gold = true
sg_node.show()
```
and unindent them so they're not inside the `if` block,
then the timer will `start()` each time gold == 2, which is what you're trying to do I think